Modules/_functoolsmodule.c

```diff
@@ -399,8 +400,20 @@ partial_repr(partialobject *pto)
         if (arglist == NULL)
             goto done;
     }
-    result = PyUnicode_FromFormat("%s(%R%U)", Py_TYPE(pto)->tp_name,
-                                  pto->fn, arglist);
+
+    mod = PyObject_GetAttrString((PyObject *)pto, "__module__");
```

Comment:
You should look up the attribute of `Py_TYPE(pto)` instead of `pto`. And you can use `_PyType_GetModuleName()` here.